qiskit/quantum_info/analysis/z2_symmetries.py

```diff
@@ -0,0 +1,485 @@
+                    built.
+                col (int): Index of the letter in the Pauli word corresponding to the single-qubit Pauli
+                    X operator.
+                idx_test (list): List of possibilities for the stacked symmetries at all other rows
+                    than row.
+                row_test (list): List of possibilities for the stacked symmetries at row.
+
+            Returns:
+                Whether or not this symmetry type should be used to build this letter of this
+                single-qubit Pauli X operator.
+            """
+            stacked_symm_idx_tests = np.array(
+                [
+                    (
+                        stacked_symm_del[row][symm_idx, col],
+                        stacked_symm_del[row][symm_idx, col + half_symm_shape],
+                    )
+                    in idx_test
+                    for symm_idx in range(symm_shape[0] - 1)
+                ]
+            )
+
+            stacked_symm_row_test = (
+                stacked_symmetries[row, col],
+                stacked_symmetries[row, col + half_symm_shape],
+            ) in row_test
+
+            return np.all(stacked_symm_idx_tests) and stacked_symm_row_test
+
+        for row in range(symm_shape[0]):
+            pauli_symmetries.append(
+                Pauli(
+                    (
+                        stacked_symmetries[row, :half_symm_shape],
+                        stacked_symmetries[row, half_symm_shape:],
+                    )
+                )
+            )
+
+            for col in range(half_symm_shape):
+                # case symmetries other than one at (row) have Z or I on col qubit
+                if _test_symmetry_row_col(row, col, test_idx["Z_or_I"], test_row["Z_or_I"]):
+                    sq_paulis.append(Pauli((np.zeros(half_symm_shape), np.zeros(half_symm_shape))))
+                    sq_paulis[row].z[col] = False
+                    sq_paulis[row].x[col] = True
+                    sq_list.append(col)
+                    break
+
+                # case symmetries other than one at (row) have X or I on col qubit
+                if _test_symmetry_row_col(row, col, test_idx["X_or_I"], test_row["X_or_I"]):
+                    sq_paulis.append(Pauli((np.zeros(half_symm_shape), np.zeros(half_symm_shape))))
+                    sq_paulis[row].z[col] = True
+                    sq_paulis[row].x[col] = False
+                    sq_list.append(col)
+                    break
+
+                # case symmetries other than one at (row)  have Y or I on col qubit
+                if _test_symmetry_row_col(row, col, test_idx["Y_or_I"], test_row["Y_or_I"]):
+                    sq_paulis.append(Pauli((np.zeros(half_symm_shape), np.zeros(half_symm_shape))))
+                    sq_paulis[row].z[col] = True
+                    sq_paulis[row].x[col] = True
+                    sq_list.append(col)
+                    break
```

Comment:
In combination with a dict along the lines of this:
```python
paulis = {
    "Z_or_I": (False, True),
    "X_or_I": (True, False),
    "Y_or_I": (True, True),
}
```
you can refactor this section further like so:
```suggestion
                for key in ("Z_or_I", "X_or_I", "Y_or_I"):
                    if _test_symmetry_row_col(row, col, test_idx[key], test_row[key]):
                        sq_paulis.append(Pauli((np.zeros(half_symm_shape), np.zeros(half_symm_shape))))
                        sq_paulis[row].z[col] = paulis[key][0]
                        sq_paulis[row].x[col] = paulis[key][1]
                        sq_list.append(col)
                        break
                else:
                    continue
                break
```

Disclaimer: I did not actually run this piece of code but I am pretty certain this is the same logic